src/figuresClasses.ts

```diff
@@ -1,11 +1,107 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'blue' | 'green';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  color: 'red' | 'blue' | 'green';
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    this.shape = 'triangle';
+    this.color = color;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Sides must be greater than 0');
+    }
+
+    const sides = [a, b, c].sort((x, y) => y - x);
+
+    if (sides[0] >= sides[1] + sides[2]) {
+      throw new Error(`Sides ${a}, ${b}, and ${c} can't form a triangle`);
```

Comment:
[CODE STYLE] - The error message here could be more descriptive. Consider something like, 'The provided side lengths of ${a}, ${b}, and ${c} do not satisfy the triangle inequality theorem and cannot form a triangle.'